test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -528,6 +504,81 @@ def setUp(self):
         self.log_end_setup()
         self.measurements = []
 
+    @classmethod
+    def populate_test_list(cls):
+        cls.test_list = [attr for attr in dir(cls)
+                         if attr.startswith('test')
+                         and callable(getattr(cls, attr))]
+
+    @classmethod
+    def skip_cray_tests(cls):
+        if not cls.mom.is_cray():
+            return
+        msg = 'capability not supported on Cray'
+        # check if we need to skip the whole test suite
+        if '__skip_on_cray__' in cls.__dict__ \
```

Comment:
you can simple do `if cls.__dict__.get('__skip_on_cray__', False):`
(same for others)